### PR TITLE
feat: Argo Workflows archive feature using RDS TDE-731

### PR DESCRIPTION
**Pull request by amfage**
#### Motivation

The aim of this change is to enable the Archived Workflows in Argo so that older workflow runs can be referenced in the future e.g. to see what was run when, what parameters were used.

#### Modification

This change provisions an AWS RDS PostgreSQL database and configures Argo Workflows to use it, so that the Archived Workflows feature is enabled.

#### Checklist

- [N/A] Tests updated
- [x] Docs updated
- [x] Issue linked in Title

### CODE REVIEW

File: infra/charts/argo.workflows.ts [hunk 1]
```diff
@@ -22,6 +24,13 @@ export interface ArgoWorkflowsProps {
    * @example "Workflows"
    */
   clusterName: string;
+  /**
+   * The Argo database endpoint
+   *
+   * @example "argodb-argodb4be14fa2-p8yjinijwbro.cmpyjhgv78aj.ap-southeast-2.rds.amazonaws.com"
+   */
+  argoDbEndpoint: string;
+  argoDbPassword: string;
```
Comment (blacha) [inline infra/charts/argo.workflows.ts:33]: could doc this, it is pretty self explanatory

File: infra/constants.ts [hunk 2]
```diff
@@ -2,11 +2,19 @@
 export const ClusterName = 'Workflows';
 /* LINZ conventional name for Argo Workflows artifact bucket */
 export const ScratchBucketName = `linz-${ClusterName.toLowerCase()}-scratch`;
+/* Argo Database Instance name */
+export const ArgoDbInstanceName = 'ArgoDb';
+/* Argo Database user */
```
Comment (blacha) [inline infra/constants.ts:7]: docs?

**Review by blacha — APPROVED** [review]

File: infra/charts/argo.workflows.ts [hunk 3]
```diff
@@ -22,6 +24,18 @@ export interface ArgoWorkflowsProps {
...
+   *
+   * @example "argodb-argodb4be14fa2-p8yjinijwbro.cmpyjhgv78aj.ap-southeast-2.rds.amazonaws.com"
+   */
+  argoDbEndpoint: string;
+  /**
+   * The Argo database password
+   *
+   * @example "eighoo5room0aeM^ahz0Otoh4aakiipo"
+   */
+  argoDbPassword: string;
```
Comment (paulfouquet) [inline infra/charts/argo.workflows.ts:38]: Where does it come from?

Comment (amfage) [reply infra/charts/argo.workflows.ts:38]: The parameter store.

Comment (paulfouquet) [reply infra/charts/argo.workflows.ts:38]: Yes but how does it arrive to the parameter store?

Comment (blacha) [reply infra/charts/argo.workflows.ts:38]: The same as all the other secrets? manually

Comment (paulfouquet) [reply infra/charts/argo.workflows.ts:38]: I guess my question is more around where do you get the password from, did you need to set up something in the AWS side

Comment (blacha) [reply infra/charts/argo.workflows.ts:38]: Yup, we should manually generate a longish password to use, We should (in a later PR) document how to rotate the password.

File: infra/cdk8s.ts [hunk 4]
```diff
@@ -27,6 +27,9 @@ async function main(): Promise<void> {
 
     // Personal access token to gain access to linz-basemaps github user
     githubPat: '/eks/github/linz-basemaps/pat',
+
+    // Argo Database connection password
+    argoDbPassword: '/eks/argo/postgres/password',
```
Comment (paulfouquet) [inline infra/cdk8s.ts:32]: That would be useful to store those secret path as constant. The same path is used in two places

**Review by paulfouquet — APPROVED** [review]

**Pull request merged**